packages/near-membrane-base/src/membrane.ts

```diff
@@ -1726,6 +1803,33 @@ export function createMembraneMarshall() {
                     } catch (e: any) {
                         throw pushErrorAcrossBoundary(e);
                     }
+                    if (
+                        result &&
+                        configurable === false &&
+                        typeof foreignCallableNonConfigurableDescriptorCallback === 'function'
+                    ) {
+                        let unsafeDesc;
+                        try {
+                            unsafeDesc = ReflectGetOwnPropertyDescriptor(target, key);
+                        } catch (e: any) {
+                            throw pushErrorAcrossBoundary(e);
+                        }
+                        if (unsafeDesc) {
+                            const descMeta = getDescriptorMeta(unsafeDesc);
+                            const { 0: descMeta0 } = descMeta;
+                            if (descMeta0 === false) {
+                                foreignCallableNonConfigurableDescriptorCallback(
+                                    descMeta0, // configurable
+                                    descMeta[1], // enumerable
+                                    descMeta[2], // writable
+                                    descMeta[3], // valuePointer
+                                    descMeta[4], // getPointer
+                                    descMeta[5] // setPointer
+                                );
+                            }
```

Comment:
☝️ `foreignCallableNonConfigurableDescriptorCallback` only when result is non-configurable :)